msal/oauth2cli/oauth2.py

```diff
@@ -147,23 +146,15 @@ def _obtain_token(  # The verb "obtain" is influenced by OAUTH2 RFC 6749
         if "token_endpoint" not in self.configuration:
             raise ValueError("token_endpoint not found in configuration")
         _headers = {'Accept': 'application/json'}
+        _headers.update(self.default_headers)
         _headers.update(headers or {})
-        resp = (post or self.session.post)(
-            self.configuration["token_endpoint"],
+        resp = (post or self.http_client.post)(self.configuration["token_endpoint"],
             headers=_headers, params=params, data=_data, auth=auth,
```

Comment:
I think, I would prefer making the necessary changes by comparing the differences and changing things only where needed. For example the dictionary update you mentioned above, I would still prefer how it is implemented here

Reply:
But, this change is necessary otherwise our implementation would generate exception when customer supplies their http layer which would not have such `auth` behavior (again, because our interface did not require such behavior). We can discuss this in meeting.

Reply:
Sorry, I dint mean this auth thing, I mean in general with the pulling changes from the oauth branch. I compared these two and  updated the auth behaviour and most of the other things from the existing implementation :)